### PR TITLE
Update NHibernate to Core v8

### PR DESCRIPTION
**Pull request by mauroservienti**
- [x] Blocked by https://github.com/Particular/NServiceBus.Gateway/pull/198

### CODE REVIEW

File: src/NServiceBus.NHibernate.Tests/ApprovalFiles/APIApprovals.Approve.approved.txt [hunk 1]
```diff
@@ -3,14 +3,6 @@
 [assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
 namespace NServiceBus.Features
 {
-    [System.ObsoleteAttribute("NHibernate gateway persistence is deprecated. Use the new NServiceBus.Gateway.Sql" +
```
Comment (timbussmann) [inline src/NServiceBus.NHibernate.Tests/ApprovalFiles/APIApprovals.Approve.approved.txt:6]: this says "will be removed in version 10", shouldn't these obsolete types/methods remain for one more major version then?

Comment (mauroservienti) [reply src/NServiceBus.NHibernate.Tests/ApprovalFiles/APIApprovals.Approve.approved.txt:6]: Ah, good catch. I'll fix that.

Comment (mauroservienti) [reply src/NServiceBus.NHibernate.Tests/ApprovalFiles/APIApprovals.Approve.approved.txt:6]: Done in https://github.com/Particular/NServiceBus.NHibernate/pull/470/commits/7cfa75274e7d9a3e94b7eb2ef31bdbf70c081406

File: src/NServiceBus.NHibernate.PersistenceTests/NServiceBus.NHibernate.PersistenceTests.csproj [hunk 2]
```diff
@@ -14,7 +14,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
     <PackageReference Include="NHibernate" Version="5.2.7" />
-    <PackageReference Include="NServiceBus.PersistenceTests.Sources" Version="7.4.2" />
+    <PackageReference Include="NServiceBus.PersistenceTests.Sources" Version="8.0.0-alpha.541" />
```
Comment (janovesk) [inline src/NServiceBus.NHibernate.PersistenceTests/NServiceBus.NHibernate.PersistenceTests.csproj:17]: This should probably by .554?

**Review by janovesk — CHANGES_REQUESTED** [review]

File: src/NServiceBus.NHibernate.PersistenceTests/NServiceBus.NHibernate.PersistenceTests.csproj (hunk 2 above)
Comment (mauroservienti) [reply src/NServiceBus.NHibernate.PersistenceTests/NServiceBus.NHibernate.PersistenceTests.csproj:17]: Yes, you're right. This was changed before we released the latest alpha. I'll fix it.

Comment (mauroservienti) [inline src/NServiceBus.NHibernate.PersistenceTests/NServiceBus.NHibernate.PersistenceTests.csproj:17]:
```suggestion
    <PackageReference Include="NServiceBus.PersistenceTests.Sources" Version="8.0.0-alpha.554" />
```

**Review by janovesk — APPROVED** [review]

**Pull request merged**